### ISSUE TITLE
Error in nested calls to procedures

### ISSUE THREAD

**Issue by scuenda** [newville/asteval#34]
First of all, great work. Thanks for doing this library.

The following code involving user defined functions does not run properly:

```
from asteval import Interpreter

aeval = Interpreter()

code="""
def fun(x):
    return x

def fun2(x):
    y=fun(x)
    return y+1

x=1
y=fun2(x)
print "should be y={}, and it is: y={}".format(x+1,y)
# should be y=2, and it is: y=1
"""
aeval(code)
```
It seems that fun2 stops running after the return of fun.

**Comment by newville** [comment]
@scuenda Hmm, I don't see that with the master branch.    

I *do* see a different problem on the master branch in that string formatting like you've done with the very convenient but easy-to-exploit `str.format` is now (as of two days ago:  585f5970bf8e909a48361a6c882a03468d18da72) disallowed.  

But for me, this::

```python
from asteval import Interpreter

aeval = Interpreter()
code="""
def fun(x):
    return x

def fun2(x):
    y=fun(x)
    print("After fun, y = %d" %  y)
    return y+1

x=1
y=fun2(x)
print("should be y=%d, and it is: y=%d" % (x+1,y))
"""
aeval(code)

```

prints
```
After fun, y = 1
should be y=2, and it is: y=2
```

Does that work for you?

**Comment by scuenda** [comment]
Yes, it does, thank you very much. 

The problem was in the version that I was using (0.9.8, installed from pip last week), not in the "format" limitation. I see that the development version has solved the issue of nested procedures. My apologies for not trying this version in the first place.

**Comment by newville** [comment]
@scuenda oh, no problem.    I think we'll need to push out a release of 0.9.9 soon.

**Comment by newville** [comment]
fixed in 0.9.9
